verilog/analysis/verilog_linter.cc

```diff
@@ -383,22 +383,68 @@ void GetLintRuleDescriptionsHelpFlag(std::ostream* os,
   }
 }
 
-void GetLintRuleDescriptionsMarkdown(std::ostream* os) {
+void GetLintRuleFile(std::ostream *os, const LinterConfiguration &config) {
+  // This rule bundle contains only a list of enabled rules. There
+  // are also no param's defined (an empty string), unless the user
+  // assigns them.
+  RuleBundle rule_bundle;
+  config.GetRuleBundle(&rule_bundle);
+
+  // Grab all the rule descriptions, so we can get default
+  // configuration and disabled rules
+  auto rule_descriptions = analysis::GetAllRuleDescriptions();
+
+  // Update the rule_bundle with default configration if none was
+  // provided and add disabled rules with default configuration
+  for (const auto &rule : rule_descriptions) {
+    // Form the rules default_configuration string
+    std::string default_configuration;
+    bool first_param = true;
+    for (const auto &param : rule.second.descriptor.param) {
+      if (!first_param) {
+        default_configuration.append(";");
+      }
+
+      default_configuration.append(std::string(param.name));
+      default_configuration.append(":");
+      default_configuration.append(param.default_value);
+
+      first_param = false;
+    }
+
+    if (rule_bundle.rules.find(rule.first) != rule_bundle.rules.end()) {
+      // Rule is enabled, add default configuration if none exists
+      if (rule_bundle.rules[rule.first].configuration.empty()) {
```

Comment:
with this access with the `[]` operator, you're doing the same look-up work multiple times.

```c++
if (auto found = rule_bundle.rules.find(rule.first); found != rule_bundle.rules.end()) {
   if (found->second.configuration.empty()) {
     found->second.configuration = default_configuration;
  }  
}
```

Reply:
Thanks. I can make that change.